### DISCUSSION TITLE
Notification permission check not working on Android

### DISCUSSION THREAD

**Discussion by DeveloperTheExplorer** [invertase/react-native-firebase#5761] [Q&A] [answered]
Following the documents provided, I'm using the following code snippet to check whether the user has allowed notifications from the app or not. This code runs perfectly on iOS and dynamically updates if I change my notification preferences of the app in the settings. However, on Android, it will always result in true even if the user has turned off notifications for the app.

```
export async function requestNotificationPermission() {
    const authStatus = await messaging().requestPermission();
    const isEnabled = authStatus === messaging.AuthorizationStatus.AUTHORIZED ||
        authStatus === messaging.AuthorizationStatus.PROVISIONAL;

    if (isEnabled) {
        console.log('Authorization status:', authStatus);
    }

    return isEnabled;
};
```

**Comment by mikehardy** [comment, marked as answer]
That is because Android requires no special permissions to display a notification, as far as I know.

The documents are explicit on the same:

> On Android, you do not need to request user permission. This method can still be called on Android devices; however, and will always resolve successfully.

https://rnfirebase.io/messaging/usage#ios---requesting-permissions

**Reply by DeveloperTheExplorer** [reply]
Is there any way to check if the user has turned off the notifications on Android?

**Reply by mikehardy** [reply]
Not in this package no. That starts to get into a very large programming API surface area that is different on different Android API levels, and is not provided by any firebase SDK APIs, so we provide no programming APIs for it here. You will want a full-featured notifications package that is made to cover device-local notification APIs, like https://notifee.app/react-native/docs/android/channels#listening-to-channel-events (note: Notifee is fully open source now, so go ahead and try it to see if it fits)

**Reply by DeveloperTheExplorer** [reply]
Thank you for the detailed and quick response :)
